scripts/dangers/keep-package-and-lock-in-sync.js

```diff
@@ -14,6 +14,117 @@
  * permissions and limitations under the License.
  */
 
+import {
+  ValidateHost,
+  ValidateHttps,
+  ValidateScheme,
+  ParseLockfile,
+} from 'lockfile-lint-api';
+
+const defaultLockfilePath = `${process.cwd()}/package-lock.json`;
+const defaultSchemes = ['npm'];
+
+export function lockFileLint(
+  pathToLockFile = defaultLockfilePath,
+  {
+    schemes = defaultSchemes,
+    scheme = false,
+    host = true,
+    protocol = true,
+    basePath = pathToLockFile,
+  } = {}
+) {
+  const options = {
+    lockfilePath: pathToLockFile,
+  };
+
+  const parser = new ParseLockfile(options);
+  const lockfile = parser.parseSync();
+  const validators = [];
+
+  if (host) {
+    validators.push(
+      [new ValidateHost({
+        packages: lockfile.object,
+      }), {
+        onSuccess() {
+          message(`[lock-files] ${basePath} has passed host validation`);
+        },
+      }]
+    );
+  }
+
+  if (protocol) {
+    validators.push(
+      [new ValidateHttps({
+        packages: lockfile.object,
+      }), {
+        onSuccess() {
+          message(`[lock-files] ${basePath} has passed protocol validation`);
+        },
+      }]
+    );
+  }
+
+  if (scheme) {
+    validators.push(
+      [new ValidateScheme({
+        packages: lockfile.object,
+      }), {
+        onSuccess() {
+          message(`[lock-files] ${basePath} has passed scheme validation`);
+        },
+      }]
+    );
+  }
+
+  validators.forEach(([validator, {
+    validate = (opts) => validator.validate(opts),
+    onWarn = (...args) => warn(...args),
+    onError = (...args) => fail(...args),
+    onSuccess = (...args) => message(...args),
+  }]) => {
+    let result;
+    try {
+      result = validate(schemes);
+    } catch (error) {
+      onWarn(error);
+    }
+
+    if (result.type === 'success') {
+      onSuccess();
+    } else if (result.type === 'error') {
+      result.errors.forEach((error) => onError(
+        `\n${error.message}\n\nat ${basePath}\n`,
+        basePath,
+        1
+      ));
+    }
+  });
+}
+
+export function runLockFilesValidation({ linting = true } = {}) {
+  const cwd = process.cwd();
+  const lockfiles = danger.git.fileMatch('**/package-lock.json');
+
+  if (linting) {
+    const {
+      modified,
+      created,
+      // TODO: pair each package-lock.with package.json, make sure they are not deleted
+      // deleted,
```

Comment:
duplication of `deleted`

Reply:
So `// deleted,` here is actually a property with the others and I just commented it out as a placeholder with the `TODO`. A reminder. I can remove if that's a blocker.